packages/popover/src/Popover.tsx

```diff
@@ -98,25 +102,35 @@ function Popover({
     adjustOnMutation,
   );
 
-  const referenceElPos = useMemo(() => getElementPosition(referenceElement), [
-    referenceElement,
-    viewportSize,
-    lastTimeRefElMutated,
-    active,
-    align,
-    justify,
-    forceUpdateCounter,
-  ]);
+  // We don't memoize these values as they're reliant on scroll positioning
+  const referenceElViewportPos = getElementViewportPosition(referenceElement);
+  const contentElViewportPos = getElementViewportPosition(contentNode);
```

Comment:
Thinking about this more, we might be able to similarly memoize these values. I'm open to thoughts on this.

Reply:
Would that require creating a useScroll hook?

Reply:
We don't need to update the component on scroll, we just need to take into account the current scroll values at the time when we're calculating the position.